### PR TITLE
feat: display settings in preferences

### LINKED ISSUE #11: feat: Display plugin settings in preferences/project preferences
Develop a tool to display the plugin settings in unity preferences or project preferences. 
Something like StansAssets.Plugins.Editor.PackageSettingsWindow but for preferences menus.

Example: 
On the screen we have "Package Manager" plugin. Based on PackageSettingsWindow we have EditorWindow, and also we display our plugin settings in the Project/Preferences menu. We should develop a standard tool to display both ways. 
image

### PR DESCRIPTION
**Pull request by ghost**
## Purpose of this PR
Implemented feature #11 

## Testing status
* No tests have been added.

### Manual testing status
Tested manually in using for the Package Manager plugin

## Comments to reviewers
**1.** Added the About page in user preferences (screenshot), which will display "about us" info and at the same time I referenced its code as a simple example of implementing preferences window. What do you think about it?

**2.** Implemented public TabController that can be useful for developing other plugins, and StansAssets.Plugins.Editor.PackageSettingsWindow<> can use it too (instead of ButtonStrip+ScrollView). It would be better to add it in another PR but PackagePreferencesWindow is dependent on it.

About page
image

Reference example
image

### CODE REVIEW

**Review by kostiantyn-koretskyi — CHANGES_REQUESTED** [review]
Few changes required

**Review by pavlo-klymentenko-legacy — CHANGES_REQUESTED** [review]
Good work! But still needs some love.
Added the video + script below with all the details.

https://github.com/StansAssets/com.stansassets.plugins-dev-kit/assets/13795723/bedec466-a737-4f23-96d4-7983efafe01a

0. Switching between Group and page
1. Changes doesnt represent when 2 winows are open
2. TabController mimics base PackageWindow
3. Registration process (attribute)? Activator.CreateInstance?
	3.1 GetPackageInfo should reused for registration (keywords, name etc)
4. PackagePreferencesWindow.cs line 31
5. CONTRIBUTING.md meta spams (foundation package update from **1.0.16** to **1.0.27**)

File: com.stansassets.plugins-dev-kit/Editor/UIToolkit/Controls/TabController.cs [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+            foreach (var tab in m_Tabs)
+            {
+                tab.Value.element.RemoveFromHierarchy();
+            }
+
+            if (!m_Tabs.Any())
+            {
+                return;
+            }
```
[the remark this answers is not in the document]
Comment (yurii-lunha-stansassets) [reply com.stansassets.plugins-dev-kit/Editor/UIToolkit/Controls/TabController.cs]: Moved

File: com.stansassets.plugins-dev-kit/Editor/UIToolkit/PreferencesWindow/AboutPreferencesWindow.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+        [SettingsProvider]
+        static SettingsProvider RegisterSettingsProvider()
+        {
+            var provider = new AboutPreferencesWindow(
+                $"{PluginsDevKitPackage.RootMenu}/Plugin Dev Kit",
+                SettingsScope.User,
+                new[] { "stan", "plugin", "dev", "kit" })
+            {
+                label = "About"
```
[the remark this answers is not in the document]
Comment (yurii-lunha-stansassets) [reply com.stansassets.plugins-dev-kit/Editor/UIToolkit/PreferencesWindow/AboutPreferencesWindow.cs]: Restructured

File: com.stansassets.plugins-dev-kit/Editor/UIToolkit/PreferencesWindow/PackagePreferencesWindow.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+            // Hide search bar from PackageSettingsWindow. In preferences we already have search bar
+            // and it's value in "searchContext" parameter
+            var searchBar = rootElement.Q<ToolbarSearchField>();
+            if (searchBar != null)
+            {
+                searchBar.style.visibility = Visibility.Hidden;
+            }
+
+            var packageInfo = GetPackageInfo();
+            rootElement.Q<Label>("display-name").text = packageInfo.displayName.Remove(0, "Stans Assets - ".Length);
```
[the remark this answers is not in the document]
Comment (yurii-lunha-stansassets) [reply com.stansassets.plugins-dev-kit/Editor/UIToolkit/PreferencesWindow/PackagePreferencesWindow.cs]: Restructured

**Comment by yurii-lunha-stansassets** [comment]
Fixed - Switching between Group and page
**Not resolved**- Changes doesn't represent when 2 windows are open
Refactored - TabController mimics base PackageWindow
Refactored- Registration process (attribute)? Activator.CreateInstance?
Refactored- GetPackageInfo should reuse for registration (keywords, name, etc)
Refactored- PackagePreferencesWindow.cs line 31
Updated - CONTRIBUTING.md meta spams (foundation package update from 1.0.16 to 1.0.27)

Please take a look at #12 - I need these changes for other features too

**Comment by pavlo-klymentenko** [comment]
All the changes merged with newer [PR](https://github.com/StansAssets/com.stansassets.plugins-dev-kit/pull/14), that's why this PR is going to be closed.
All credits to @yurii-lunha-stansassets for updates!

**Pull request closed without merging**